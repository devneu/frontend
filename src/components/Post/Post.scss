.post {
  .tag-post {
    color: white;
    font-weight: 1000;
    border-radius: 5px;
    margin-top: 10px;
  }

  .ant-image, {
    border-radius: 5px;
    width: 100%;
  }

  .text-post {
    font-size: 20px;
  }
}
