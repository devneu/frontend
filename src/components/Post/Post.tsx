import React from 'react';
import { PageHeader, Typography, Tag, Image } from 'antd';
import { useHistory, useParams } from 'react-router-dom';
import Editor from 'rich-markdown-editor';
import { createMockPosts } from '../../mock/mockPost';
import './Post.scss';

const { Title, Text, Paragraph } = Typography;

const Post = () => {
  const history = useHistory();
  const { postId } = useParams<any>();
  const currentPost = createMockPosts().find((el) => el.id === +postId);

  if (!currentPost) {
    history.push('/page-not-found');
    return null;
  }

  return (
    <div className="post">
      <PageHeader>
        <Title>Welcome to Blog Post!</Title>
        <Text type="secondary" italic>
          {currentPost.date}
        </Text>
        <div>
          {currentPost.tag.map((tag) => (
            <Tag className="tag-post" color="#6c757d">
              {tag}
            </Tag>
          ))}
        </div>
      </PageHeader>
      <Image className="img-post" src={currentPost.img} />
      <Paragraph className="text-post">{currentPost.text}</Paragraph>
      <Editor defaultValue={currentPost.articleText} readOnly className="text-post" />
    </div>
  );
};

export default Post;
