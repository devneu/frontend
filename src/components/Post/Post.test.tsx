import React from 'react';
import renderer from 'react-test-renderer';
import ReactRouter from 'react-router';
import Post from './Post';

jest.mock('react-markdown', () => () => 'MockReactMarkDown');

describe('Post', () => {
  let useParamsSpy: jest.SpyInstance;
  let useHistorySpy: jest.SpyInstance;

  beforeEach(() => {
    useParamsSpy = jest.spyOn(ReactRouter, 'useParams').mockReturnValue({});
    useHistorySpy = jest.spyOn(ReactRouter, 'useHistory').mockReturnValue({ push: jest.fn() } as any);
  });

  afterEach(() => {
    useParamsSpy?.mockRestore();
    useHistorySpy?.mockRestore();
  });

  it('Should successfully render', () => {
    const wrapper = renderer.create(<Post />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
