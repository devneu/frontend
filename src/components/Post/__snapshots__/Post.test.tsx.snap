// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Should successfully render 1`] = `null`;
